safe-global/safe-wallet-web
# feat: recovery transaction processing state

## What it solves

Resolves [missing "Processing..." state for recovery transactions](https://www.notion.so/safe-global/Do-we-have-a-processing-relaying-state-instead-of-awaiting-confirmation-for-the-Account-recovery-5a07f9ed7cde429ba976fe7ad16e5f59)

## How this PR fixes it

A new recovery-focused event bus has been added that matches that of the one for "standard" transactions and messages. Using that, a new pending map is updated according to the dispatched events and parsed for the current status of a recovery attempt.

## How to test it

1. Queue a recovery attempt.
2. Execute/cancel the attempt and observe the new processing status.

Note: data loading should also be tested - recovery state updates when a proposal is executed, cancelled and/or settings are changed.

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/c1a15dc7-304f-4524-9499-86602d5b989b)

## Checklist
